### ISSUE TITLE
SAWCore `error` function calls `panic`

### ISSUE THREAD

**Issue by robdockins** [GaloisInc/saw-core#59]
This is probably not what we want.  Legitimate user input can lead to SAWCore `error`.  For example, using the `at` function with an out-of-bound index calls `error`, and this should clearly be a verification failure rather than a panic.

**Comment by brianhuffman** [comment]
Yeah, we should probably get around to deciding what partial functions actually mean in saw-core.

**Comment by brianhuffman** [comment]
PR #61 is a band-aid fix for the problem, although it might be enough for now. In the long run, we need to properly support partial values in the saw-core evaluation backends.
